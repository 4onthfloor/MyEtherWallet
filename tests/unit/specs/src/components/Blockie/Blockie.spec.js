import Blockie from '@/components/Blockie';
import { mount } from '@vue/test-utils';

describe('Blockie.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(Blockie, {
      propsData: {
        address: '0xDECAF9CD2367cdbb726E904cD6397eDFcAe6068D'
      }
    });
  });

  afterAll(() => {
    wrapper.destroy();
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correct contents', () => {
    expect(wrapper.element.style['background-image']).toEqual(
      'url(data:image/png;base64,00)'
    );
  });

  it('dismounts', () => {
    wrapper.destroy();
    expect(wrapper.exists()).toBe(false);
  });
});
