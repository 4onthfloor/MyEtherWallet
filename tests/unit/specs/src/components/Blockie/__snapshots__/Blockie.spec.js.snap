// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockie.vue renders correctly 1`] = `<div class="address-identicon" style="width: 64px; height: 64px; background-image: url(data:image/png;base64,00);"></div>`;
